.gitignore

```diff
@@ -1,30 +1,46 @@
-/env*
-*.pyc
-*.egg-info/
-tmp/
+# gitignore cheatsheet:
+#
+# Matches with no '/'s automatically match files in sub-dirs.
+# Matches starting with a '/' only match in the top-level directory.
+# Matches ending with a '/' only match directories.
+# ** can be used to match multiple levels of directory.
+
+# Global glob patterns.
 *.log
-/.coverage
-/htmlcov
-/.dir-locals.el
-/.tox
-/docs/build
-/.idea
-/.eggs
-/.hypothesis
-/coverage.xml
-build
-dist
-site-packages
-.noseids
-nosetests.xml
-version.txt
-bin
-local-utils
-python/version.py
-.go-pkg-cache
-vendor
-bin
 *.coverprofile
 *.pb.go
+/.noseids
+/nosetests.xml
+
+# Build directories.
+/bin/
+/docker-image/bin/
+/build/
+/dist/
+.go-pkg-cache/
+/.glide/
+/vendor/
+
+# IDE files.
+/.idea/
+
+# Patterns to match files from older releases (prevents lots of
+# untracked files from showing up when switching from a maintenance branch.
+
+# Python builds/UT data.
+.hypothesis/
+.coverage
 *_pb2.py
-.glide
+/env*
```

Comment:
This one looks dangerous, e.g. would accidentally ignore a new 'environ' go package.  Can it be more specific?